spdm.rflx

```diff
@@ -0,0 +1,1487 @@
+         Base_Hash_Algo_TPM_ALG_SHA_512 : Boolean;
+         Base_Hash_Algo_TPM_ALG_SHA_384 : Boolean;
+         Base_Hash_Algo_TPM_ALG_SHA_256 : Boolean;
+         Base_Hash_Algo_Reserved_2 : Reserved_24;
+         Reserved_2 : Reserved_64;
+         Reserved_3 : Reserved_32;
+         Ext_Asym_Count : Ext_Asym_Count;
+         Ext_Hash_Count : Ext_Hash_Count;
+         Reserved_4 : Reserved_16;
+         Ext_Asym : Ext_Algs
+            with Size => 32 * Ext_Asym_Count;
+         Ext_Hash : Ext_Algs
+            with Size => 32 * Ext_Hash_Count;
+      end message;
+
+   type Negotiate_Algorithms_Request_1_1 is
+      message
+         Req_Alg_Struct_Count : Alg_Struct_Count;
+         Param_2 : Reserved_8;
+         Length_1 : Length;
+         Length_2 : Length
+            then Measurement_Specification_Reserved
+               if Length_2 * 256 + Length_1 <= 128;
+         Measurement_Specification_Reserved : Reserved_7;
+         Measurement_Specification_DMTF : Boolean;
+         Reserved_1 : Reserved_8;
+         Base_Asym_Algo_TPM_ALG_ECDSA_ECC_NIST_P384 : Boolean;
+         Base_Asym_Algo_TPM_ALG_RSAPSS_4096 : Boolean;
+         Base_Asym_Algo_TPM_ALG_RSASSA_4096 : Boolean;
+         Base_Asym_Algo_TPM_ALG_ECDSA_ECC_NIST_P256 : Boolean;
+         Base_Asym_Algo_TPM_ALG_RSAPSS_3072 : Boolean;
+         Base_Asym_Algo_TPM_ALG_RSASSA_3072 : Boolean;
+         Base_Asym_Algo_TPM_ALG_RSAPSS_2048 : Boolean;
+         Base_Asym_Algo_TPM_ALG_RSASSA_2048 : Boolean;
+         Base_Asym_Algo_Reserved_1 : Reserved_7;
+         Base_Asym_Algo_TPM_ALG_ECDSA_ECC_NIST_P521 : Boolean;
+         Base_Asym_Algo_Reserved_2 : Reserved_16;
+         Base_Hash_Algo_Reserved_1 : Reserved_2;
+         Base_Hash_Algo_TPM_ALG_SHA3_512 : Boolean;
+         Base_Hash_Algo_TPM_ALG_SHA3_384 : Boolean;
+         Base_Hash_Algo_TPM_ALG_SHA3_256 : Boolean;
+         Base_Hash_Algo_TPM_ALG_SHA_512 : Boolean;
+         Base_Hash_Algo_TPM_ALG_SHA_384 : Boolean;
+         Base_Hash_Algo_TPM_ALG_SHA_256 : Boolean;
+         Base_Hash_Algo_Reserved_2 : Reserved_24;
+         Reserved_2 : Reserved_64;
+         Reserved_3 : Reserved_32;
+         Ext_Asym_Count : Ext_Asym_Count;
+         Ext_Hash_Count : Ext_Hash_Count;
+         Reserved_4 : Reserved_16;
+         Ext_Asym : Ext_Algs
+            with Size => 32 * Ext_Asym_Count;
+         Ext_Hash : Ext_Algs
+            with Size => 32 * Ext_Hash_Count
+            then Req_Alg_Structs
+               --  Length is the total length of the message, we have to subtract the common header
+               --  of Request.
+               with Size => 8 * (Length_2 * 256 + Length_1) - 16 - (Ext_Hash'Last - Req_Alg_Struct_Count'First + 1)
+                  if 8 * (Length_2 * 256 + Length_1) >= 16 + (Ext_Hash'Last - Req_Alg_Struct_Count'First + 1);
+         Req_Alg_Structs : Req_Alg_Structs;
+         -- ISSUE: Componolit/systematization-binary-schemas#8
+         -- The number of elements should also match Req_Alg_Struct_Count
```

Comment:
Replace by Componolit/RecordFlux#51

Reply:
Fixed in https://github.com/Componolit/RecordFlux-specifications/pull/93/commits/7c1908a0db7eece9cbbb987fa14b0c9d9e4ec91f.